macros/cross_db_utils/last_day.sql

```diff
@@ -0,0 +1,37 @@
+/*
+This function has been tested with dateparts of month and quarters. Further
+testing is required to validate that it will work on other dateparts.
+*/
+
+{% macro last_day(date, datepart) %}
+  {{ adapter_macro('dbt_utils.last_day', date, datepart) }}
+{% endmacro %}
+
+
+{% macro default__last_day(date, datepart) -%}
+    cast(
+        {{dbt_utils.dateadd('day', '-1',
+        dbt_utils.dateadd(datepart, '1', dbt_utils.date_trunc(datepart, date))
+        )}}
+        as date)
+{%- endmacro %}
+
+
+{% macro postgres__last_day(date, datepart) -%}
+
+    {%- if datepart == 'quarter' -%}
+
+    {{ exceptions.raise_compiler_error(
+        "dbt_utils.last_day is not supported for datepart 'quarter' on this adapter") }}
+
+    {%- else -%}
+
+    cast(
```

Comment:
Is this code the same as the code in the default `last_day` macro? If so: we should probably just call that macro here instead of duplicating the logic, right?

Reply:
yep! that's exactly it.